src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,207 @@
+        currentNode.value = value;
+        return oldValue;
     }
-
+    
     @Override
     public T remove(int index) {
+        checkIncludeIndex(index);
+        if (size == 1 && index == 0) {
+            head = null;
+            tail = null;
+            size = 0;
+        }
+        int i = 0;
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (i == index) {
+                Node<T> nextNode = currentNode.next;
+                Node<T> prevNode = currentNode.prev;
+                if (prevNode == null) {
+                    nextNode.prev = null;
+                    head = nextNode;
+                    size--;
+                    return currentNode.value;
+                }
+                if (nextNode == null) {
+                    prevNode.next = null;
+                    tail = prevNode;
+                    size--;
+                    return currentNode.value;
+                }
+                nextNode.prev = prevNode;
+                prevNode.next = nextNode;
+                size--;
+                return currentNode.value;
+            }
+            i++;
+            currentNode = currentNode.next;
+        }
         return null;
     }
-
+    
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.value == object
+                    || (currentNode.value != null && currentNode.value.equals(object))) {
+                if (size == 1) {
+                    head = null;
+                    tail = null;
+                    size = 0;
+                    return true;
+                }
+                Node<T> nextNode = currentNode.next;
+                Node<T> prevNode = currentNode.prev;
+                if (prevNode == null) {
+                    nextNode.prev = null;
+                    head = nextNode;
+                    size--;
+                    return true;
+                }
+                if (nextNode == null) {
+                    prevNode.next = null;
+                    tail = prevNode;
+                    size--;
+                    return true;
+                }
+                nextNode.prev = prevNode;
+                prevNode.next = nextNode;
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
-
+    
     @Override
     public int size() {
-        return 0;
+        return size;
     }
-
+    
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+    
+    public T getFirst() {
+        return head.value;
+    }
+    
+    public T getLast() {
+        return tail.value;
+    }
+    
+    private Node<T> getNode(int index) {
+        Node<T> currentNode = head;
+        int i = 0;
+        while (currentNode != null) {
+            if (i == index) {
```

Comment:
What if your node would be the last before tail? Probably in some cases, it's better to start from tail?